src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -305,12 +305,51 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
             return dbResultSet;
         }
 
+        /// <inheritdoc />
+        public override SqlCommand PrepareDbCommand(
+            SqlConnection conn,
+            string sqltext,
+            IDictionary<string, DbConnectionParam> parameters,
+            HttpContext? httpContext,
+            string dataSourceName)
+        {
+            SqlCommand cmd = conn.CreateCommand();
+            cmd.CommandType = CommandType.Text;
+
+            // Add query to send user data from DAB to the underlying database to enable additional security the user might have configured
+            // at the database level.
+            string sessionParamsQuery = GetSessionParamsQuery(httpContext, parameters, dataSourceName);
+
+            cmd.CommandText = sessionParamsQuery + sqltext;
+            if (parameters is not null)
+            {
+                foreach (KeyValuePair<string, DbConnectionParam> parameterEntry in parameters)
+                {
+                    SqlParameter parameter = cmd.CreateParameter();
+                    parameter.ParameterName = parameterEntry.Key;
+                    parameter.Value = parameterEntry.Value.Value ?? DBNull.Value;
+                    PopulateDbTypeForParameter(parameterEntry, parameter);
```

Comment:
Most of this method seems to be a duplicate of the base class' method definition. Can't just providing an overridden implementation for `PopulateDbTypeForParameter()` method suffice?

Reply:
No, because we need `SqlCommand` and not `DbCommand` which is there in the base implementation.